src/Achieve.Aspire.AzureProvisioning/Bicep/Batch/BatchAccountResource.cs

```diff
@@ -46,7 +46,7 @@ public BatchAccountResource(string name) : base(resourceType)
 
     protected override void ValidateResourceType()
     {
-        if (Name.MatchesConstraints(3, 24, StringExtensions.CharacterClass.LowercaseLetter))
+        if (!AccountName.MatchesConstraints(3, 24, StringExtensions.CharacterClass.LowercaseLetter))
```

Comment:
Ah yes, this "Name" has tripped me up a couple of times too given the ambiguity, something else to look at for the internal API.